src/TopDownProteomics/Chemistry/InMemoryElementProvider.cs

```diff
@@ -29,7 +29,7 @@ public InMemoryElementProvider(IElement[] elements)
         private void IndexElements(IElement[] elements)
         {
             int capacity = elements.Max(x => x.AtomicNumber);
-            _by_atomic_number = new IElement[capacity];
+            _by_atomic_number = new IElement[capacity + 1];
             _by_symbol = new Dictionary<string, IElement>(capacity);
```

Comment:
Should up the initial capacity here too.

Reply:
Actually, I'm wrong. This shouldn't be capacity at all, should just be the length of the elements array